webpack/components/SelectOrg/SelectOrgReducer.js

```diff
@@ -0,0 +1,36 @@
+import Immutable from 'seamless-immutable';
+import {
+  GET_ORGANIZATIONS_LIST_SUCCESS,
+  GET_ORGANIZATIONS_LIST_REQUEST,
+  GET_ORGANIZATIONS_LIST_FAILURE,
+  CHANGE_CURRENT_ORGANIZATION_SUCCESS,
+} from '../../redux/consts';
+
+const initialState = Immutable({ loading: false });
+export default (state = initialState, action) => {
+  const { payload } = action;
+
+  switch (action.type) {
+    case GET_ORGANIZATIONS_LIST_REQUEST:
+      return state.set('loading', true);
+
+    case GET_ORGANIZATIONS_LIST_SUCCESS:
+      return state
+        .set('list', payload.results.reduce((obj, item) => {
+          obj[item.name] = item.id; // eslint-disable-line no-param-reassign
```

Comment:
Why is this reassignment necessary?

Reply:
Because we need a different structure for the select component.